### ISSUE TITLE
Money printer / blood pact crashes the game

### ISSUE THREAD

**Issue by gobbi9** [notmario/MoreFluff#1]
Playing the following hand with money printer crashes the game. I have a lot of mods installed, it might be a conflict with another mod, I did not test it with this mod alone:

<img width="478" alt="hand" src="https://github.com/notmario/MoreFluff/assets/5932540/ec0c6eea-d6c9-48a4-bef8-11ae5ea17c5d">
<img width="370" alt="error" src="https://github.com/notmario/MoreFluff/assets/5932540/fe32c638-1e3a-4889-8c90-0e3686065f54">

Installed mods:
<img width="538" alt="mods" src="https://github.com/notmario/MoreFluff/assets/5932540/f2e9574e-57ce-43db-966e-dc57648832c3">

**Comment by gobbi9** [comment]
Trying to buy blood pact also crashes the game:

![image](https://github.com/notmario/MoreFluff/assets/5932540/588268b9-9312-42fd-9374-7d0e4a46a0fc)

**Comment by gobbi9** [comment]
For blood pact, adding a null check at the beginning of the calculate function seems to fix it

```lua
    SMODS.Jokers.j_mf_bloodpact.calculate = function(self, context)
        if SMODS.end_calculate_context(context) and context.full_hand ~= nil then
           ...
```

**Comment by Cr4zZyBipBiip** [comment]
> Playing the following hand with money printer crashes the game. I have a lot of mods installed, it might be a conflict with another mod, I did not test it with this mod alone:
> 
> <img alt="hand" width="478" src="https://private-user-images.githubusercontent.com/5932540/316307284-ec0c6eea-d6c9-48a4-bef8-11ae5ea17c5d.PNG?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE2MzA1ODUsIm5iZiI6MTcxMTYzMDI4NSwicGF0aCI6Ii81OTMyNTQwLzMxNjMwNzI4NC1lYzBjNmVlYS1kNmM5LTQ4YTQtYmVmOC0xMWFlNWVhMTdjNWQuUE5HP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDMyOCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAzMjhUMTI1MTI1WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9M2I0OWUyODNjMzVjMTIxZTE0OGNmMTcyOTU3MDA2OWQxZWM5ZDIxNjU0MDQ0ZGMwNmJjMmFmMzA3ZmMzNWFlZSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.9eImsV8Np75meeSHLSa2nso41Y_Uo_xGxcl4lVQp2cU"> <img alt="error" width="370" src="https://private-user-images.githubusercontent.com/5932540/316307281-fe32c638-1e3a-4889-8c90-0e3686065f54.PNG?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE2MzA1ODUsIm5iZiI6MTcxMTYzMDI4NSwicGF0aCI6Ii81OTMyNTQwLzMxNjMwNzI4MS1mZTMyYzYzOC0xZTNhLTQ4ODktOGM5MC0wZTM2ODYwNjVmNTQuUE5HP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDMyOCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAzMjhUMTI1MTI1WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YmQzOWYxMjhhNWRlMmM1MzBkMjNhMjgzODZlMWY4NTg2NWZlMzI0ODUyNmQxM2ZjN2Y2OGZkNGU0Njg5ZWU5MSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.fj00vv-REARcXscl4QtBOIsF_IFs_Gmsuf0Wm1woAlg">
> Installed mods: <img alt="mods" width="538" src="https://private-user-images.githubusercontent.com/5932540/316307280-f2e9574e-57ce-43db-966e-dc57648832c3.PNG?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE2MzA1ODUsIm5iZiI6MTcxMTYzMDI4NSwicGF0aCI6Ii81OTMyNTQwLzMxNjMwNzI4MC1mMmU5NTc0ZS01N2NlLTQzZGItOTY2ZS1kYzU3NjQ4ODMyYzMuUE5HP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDMyOCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAzMjhUMTI1MTI1WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MGJjOTk4NDUyODVmNGY4MjJlYjhkZmNmNWY0NjgyZmMyYmMxOTNkM2ZmN2Y0NzhiNjVlYjE4ODE0NTY3NTE1MyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.NXRV3BVFSCWoE65pUdtmVOcEtN11vSEVuuIUMIvbBiQ">

Fixed in PR
https://github.com/notmario/MoreFluff/pull/2

**Comment by notmario** [comment]
fixed for 1.2
